source/slang/slang.cpp

```diff
@@ -3842,6 +3854,7 @@ void EndToEndCompileRequest::setLineDirectiveMode(SlangLineDirectiveMode mode)
 {
     // TODO: validation
     getBackEndReq()->lineDirectiveMode = LineDirectiveMode(mode);
+    m_linkage->setLineDirectiveMode(LineDirectiveMode(mode));
```

Comment:
Okay, I see. It seems like the existing line-directive mode is coming through from the end-to-end (command line) request, and thus maps most directly to a linkage-wide setting.

It would probably be good to have a later change make this a per-target-req setting instead, because that would seem to be the appropriate granularity for setting it, *especially* since the line directive mode is really part of the implementation of "debug information" for a particular target.